### ISSUE TITLE
Column start (3) greater than line length (2)

### ISSUE THREAD

**Issue by BlaM** [AtomLinter/linter-php#103]
In one particular source code file (sorry can't share) I see this error message whenever I start creating an if-block

    Error: Column start (3) greater than line length (2)
        at Object.module.exports.Helpers.rangeFromLineNumber (file:///C:/Users/ddeobald/.atom/packages/linter-php/node_modules/atom-linter/lib/helpers.coffee:73:17)
        at file:///C:/Users/ddeobald/.atom/packages/linter-php/lib/main.coffee:58:30
        at process._tickCallback (node.js:368:9)

The error appears when I'm at this point in typing:

    if ($x)

(i.e. no "then" action after the if)

So far I'm trying to reproduce this in another file. Not sure what is actually causing this. It won't appear in any other file. The error is only triggered if I put that "if ($x)" inside a member function of a class in that specific file - so my best guess is that some other content of that file is triggering this. 

NO Error appears if I put it in a function outside of a class.

I will have to investigate more, but maybe the error message is enough to start looking into this?

**Comment by Arcanemagus** [comment]
If you save the file at the exact state that triggers this, what does `php -l foo.php` show?

**Comment by Arcanemagus** [comment]
Can you do the following:
* Trigger the error and note _which_ `helpers.coffee` it was triggered from (many linter plugins use this and Atom picks one at basically random)
* Put a breakpoint on [this line](https://github.com/AtomLinter/atom-linter/blob/master/lib/helpers.coffee#L69) in the `helpers.cofee` that is currently being used
* Trigger the error again
* Get the value of the following in the console:
```
textEditor.indentationForBufferRow(lineNumber)
textEditor.getTabLength()
buffer.lineLengthForRow(lineNumber)
```

----
The only way for this to trigger is somehow the way `helpers.rangeFromLineNumber()` is determining the start of the line ends up being after the _end_ of the line since `linter-php` doesn't specify a column when calling it.

**Comment by BlaM** [comment]
I'll try on monday when I'm back in the office. 

Can't promise anything. I'm not an Atom developer, I'm just a user - and while I could probably help locate the bug with JavaScript I don't really care to learn coffee-script. 

 [ For reference: https://github.com/AtomLinter/atom-linter/issues/63 ]

**Comment by Arcanemagus** [comment]
Marking as duplicate of https://github.com/AtomLinter/atom-linter/issues/63, I'm not sure why you are only now seeing this as before it would have been crashing inside an Atom function, we simply added a check to catch this failure before calling the Atom function.

Can you see if your file mixes tabs and spaces for indentation?

**Comment by BlaM** [comment]
Just for the records, yes - I have mixed tabs and spaces. The file I'm working on is quite old and some vintage code -> mixed styles.
